addons/website/static/src/snippets/s_map/000.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+import publicWidget from 'web.public.widget';
+
+publicWidget.registry.Map = publicWidget.Widget.extend({
+    selector: '.s_map',
+
+    /**
+     * @override
+     */
+    start() {
+        if (!this.el.querySelector('.s_map_embedded')) {
+            // The iframe is not found inside the snippet. This is probably due
+            // the sanitization of a field during the save, like in a product
+            // description field.
+            // In such cases, reconstruct the iframe.
+            const dataset = this.$target[0].dataset;
+            if (dataset.mapAddress) {
+                const url = 'https://maps.google.com/maps?q=' + encodeURIComponent(dataset.mapAddress)
+                    + '&t=' + encodeURIComponent(dataset.mapType)
+                    + '&z=' + encodeURIComponent(dataset.mapZoom)
+                    + '&ie=UTF8&iwloc=&output=embed';
+                const iframeEl = document.createElement('iframe');
+                iframeEl.classList.add('s_map_embedded', 'o_not_editable');
+                iframeEl.setAttribute('width', '100%');
+                iframeEl.setAttribute('height', '100%');
+                iframeEl.setAttribute('frameborder', '0');
+                iframeEl.setAttribute('scrolling', 'no');
+                iframeEl.setAttribute('marginheight', '0');
+                iframeEl.setAttribute('marginwidth', '0');
+                iframeEl.setAttribute('src', url);
+                this.el.querySelector('.s_map_color_filter').before(iframeEl);
+            }
+        }
+        return this._super(...arguments);
+    },
+});
+
+export default {
+    Map: publicWidget.registry.Map,
+};
```

Comment:
Unless I am mistaken, the common way would be to export the widget only (not an object containing the widget). Detail though, but would mean less code so.. :shrug: